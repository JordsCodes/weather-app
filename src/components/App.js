import React, { useState, useEffect } from "react";
import axios from "axios";
import LocationDetails from "./LocationDetails";
import ForecastSummaries from "./ForecastSummaries";
import ForecastDetails from "./ForecastDetails";
import getForecast from "../requests/getForecast";

function App() {
  const [selectedDate, setSelectedDate] = useState(0);
  const [forecasts, setForecasts] = useState([]);
  const [location, setLocation] = useState({ city: "", country: "" });

  useEffect(() => {
    getForecast(setSelectedDate, setForecasts, setLocation);
  }, []);

  const selectedForecast = forecasts.find((item) => item.date === selectedDate);
  const { city, country } = location;
  const handleForecastSelect = (date) => {
    setSelectedDate(date);
  };
  return (
    <div className="weather-app">
      <LocationDetails city={city} country={country} />
      <ForecastSummaries forecasts={forecasts} onForecastSelect={handleForecastSelect} />
      {selectedForecast && (<ForecastDetails forecast={selectedForecast} />)}
    </div>
  );
}

export default App;
