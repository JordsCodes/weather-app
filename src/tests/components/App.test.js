import React from "react";
import axios from "axios";
import { render, screen, waitFor } from "@testing-library/react";
import App from "../../components/App";

describe("App", () => {
  const forecast = {
    location: {
      city: "Manchester",
      country: "UK",
    },
    forecasts: [
      {
        date: 1525046400000,
        temperature: {
          max: 11,
          min: 4,
        },
        wind: {
          speed: 13,
          direction: "s",
        },
        humidity: 30,
        description: "Clear",
        icon: "800",
      }],
  };

  it("renders correctly", async () => {
    jest.spyOn(axios, "get").mockResolvedValue({ data: forecast });
    const { asFragment } = render(
      <App />,
    );
    await waitFor(() => {
      const h1Element = screen.getByText(/Manchester, UK/i);

      expect(h1Element).toBeInTheDocument();
    });

    expect(asFragment()).toMatchSnapshot();
  });
});
