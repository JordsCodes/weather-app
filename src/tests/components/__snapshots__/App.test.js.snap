// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
<DocumentFragment>
  <div
    class="weather-app"
  >
    <h1>
      Manchester, UK
    </h1>
    <div
      class="forecast-summaries"
    >
      <div
        class="forecast-summary"
        data-testid="forecast-summary"
      >
        <div
          class="forecast-summary_date"
        >
          Mon Apr 30 2018
        </div>
        <div
          class="forecast-summary_icon"
          data-testid="forecast-icon"
        >
          <img
            alt="thunderstorm"
            src="../images/800.png"
          />
        </div>
        <div
          class="forecast-summary_temperature"
        >
          11°C
        </div>
        <div
          class="forecast-summary_description"
        >
          Clear
        </div>
        <button
          type="button"
        >
          More details
        </button>
      </div>
      <div
        class="forecast-summary"
        data-testid="forecast-summary"
      >
        <div
          class="forecast-summary_date"
        >
          Tue May 01 2018
        </div>
        <div
          class="forecast-summary_icon"
          data-testid="forecast-icon"
        >
          <img
            alt="thunderstorm"
            src="../images/200.png"
          />
        </div>
        <div
          class="forecast-summary_temperature"
        >
          13°C
        </div>
        <div
          class="forecast-summary_description"
        >
          Stormy
        </div>
        <button
          type="button"
        >
          More details
        </button>
      </div>
      <div
        class="forecast-summary"
        data-testid="forecast-summary"
      >
        <div
          class="forecast-summary_date"
        >
          Wed May 02 2018
        </div>
        <div
          class="forecast-summary_icon"
          data-testid="forecast-icon"
        >
          <img
            alt="thunderstorm"
            src="../images/600.png"
          />
        </div>
        <div
          class="forecast-summary_temperature"
        >
          1°C
        </div>
        <div
          class="forecast-summary_description"
        >
          Heavy Snow
        </div>
        <button
          type="button"
        >
          More details
        </button>
      </div>
      <div
        class="forecast-summary"
        data-testid="forecast-summary"
      >
        <div
          class="forecast-summary_date"
        >
          Thu May 03 2018
        </div>
        <div
          class="forecast-summary_icon"
          data-testid="forecast-icon"
        >
          <img
            alt="thunderstorm"
            src="../images/700.png"
          />
        </div>
        <div
          class="forecast-summary_temperature"
        >
          20°C
        </div>
        <div
          class="forecast-summary_description"
        >
          Tornado
        </div>
        <button
          type="button"
        >
          More details
        </button>
      </div>
      <div
        class="forecast-summary"
        data-testid="forecast-summary"
      >
        <div
          class="forecast-summary_date"
        >
          Fri May 04 2018
        </div>
        <div
          class="forecast-summary_icon"
          data-testid="forecast-icon"
        >
          <img
            alt="thunderstorm"
            src="../images/700.png"
          />
        </div>
        <div
          class="forecast-summary_temperature"
        >
          25°C
        </div>
        <div
          class="forecast-summary_description"
        >
          Hazy
        </div>
        <button
          type="button"
        >
          More details
        </button>
      </div>
    </div>
    <div
      class="forecast-details"
    >
      <ul>
        <li>
          Date:Mon Apr 30 2018
        </li>
        <li>
          Temperature High/Low:114
        </li>
        <li>
          Wind Speed/Direction:13s
        </li>
        <li>
          Humidity:30%
        </li>
      </ul>
    </div>
  </div>
</DocumentFragment>
`;
